src/librustc_middle/ty/codec.rs

```diff
@@ -60,6 +62,7 @@ where
     E: TyEncoder,
     M: for<'b> Fn(&'b mut E) -> &'b mut FxHashMap<T, usize>,
     T: EncodableWithShorthand,
+    usize: TryFrom<<T::Variant as DiscriminantKind>::Discriminant>,
```

Comment:
Please change this to `T::Variant: DiscriminantKind<Discriminant = isize>` and keep the original code in the body.
This is an internal function and its performance shouldn't change.

Reply:
this sadly does not work, as `DiscriminantKind::Discriminantt` is `u64` during bootstrap. While this could be solved using a small local trait I don't know if this worth it. I can add a fixme if you want though.